node/src/test/java/io/airlift/node/TestNodeModule.java

```diff
@@ -78,6 +82,7 @@ public void testFullConfig()
         String binarySpec = "binary";
         String configSpec = "config";
         String publicAddress = "public";
+        File annotationFile = new File(requireNonNull(getClass().getClassLoader().getResource("annotations.properties")).getFile());
```

Comment:
requireNonNull -> use guava `Resources`
`URL.getFile()` -> this will pass for resource URLs like `jar:...`, but you want it to fail

use `new File(url.toURI())`

Reply:
This is a test and the resource name is hardcoded, not provided by users. I'm not sure if I understand the requested change, can you add a code suggestion?

Reply:
```
File annotationFile = new File([Resources.]getResource("annotations.properties").toURI());
```